test/connector/tcp/mssql/client_request.go

```diff
@@ -0,0 +1,192 @@
+package mssqltest
+
+import (
+	"net"
+	"strconv"
+
+	"github.com/cyberark/secretless-broker/internal/log"
+	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp"
+	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp/mssql"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+type clientRequest struct {
+	params dbConfigParams
+	query  string
+}
```

Comment:
Naming:

As I understand it, this is just a query intended to be executed on a particular database.

As such, I'd call it `DbSpecificQuery`.

Similarly, `dbConfigParams` is defined as:

```go
type dbConfigParams struct {
	Database string
	// This is in relation to what is generally referred to as Application Intent.
	// It can only take 2 values, ReadWrite or ReadOnly.
	ReadOnly bool
}
```
Really, it's just "a database", with this one extra setting.  I would suggest `DatabaseType` or similar, since it's primary function is to identify the database, and not to provide configuration, and there are no parameters in sight.